src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,314 @@
+ * @param {import('../types.d.ts').HierarchizedEntry} entry - The AST entry.
+ * @param {HeadingMetadataParent} head - The head node of the entry.
+ * @returns {import('../types.d.ts').Section} The created section.
+ */
+function createSection(entry, head) {
+  return {
+    textRaw: transformNodesToString(head.children),
+    name: head.data.name,
+    type: head.data.type,
+    meta: createMeta(entry),
+    introduced_in: entry.introduced_in,
+  };
+}
+
+/**
+ * Parses a list item to extract properties.
+ * @param {import('mdast').ListItem} child - The list item node.
+ * @param {import('../types.d.ts').HierarchizedEntry} entry - The entry containing raw content.
+ * @returns {import('../types.d.ts').List} The parsed list.
+ */
+function parseListItem(child, entry) {
+  const current = {};
+
+  /**
+   * Extracts raw content from a node based on its position.
+   * @param {import('mdast').BlockContent} node
+   * @returns {string}
+   */
+  const getRawContent = node =>
+    entry.rawContent.slice(
+      node.position.start.offset,
+      node.position.end.offset
+    );
+
+  /**
+   * Extracts a pattern from text and assigns it to the current object.
+   * @param {string} text
+   * @param {RegExp} pattern
+   * @param {string} key
+   * @returns {string}
+   */
+  const extractPattern = (text, pattern, key) => {
+    const [, match] = text.match(pattern) || [];
+    if (match) {
+      current[key] = match.trim().replace(/\.$/, '');
+      return text.replace(pattern, '');
+    }
+    return text;
+  };
+
+  // Combine and clean text from child nodes, excluding nested lists
+  current.textRaw = child.children
+    .filter(node => node.type !== 'list')
+    .map(getRawContent)
+    .join('')
+    .replace(/\s+/g, ' ')
+    .replace(/<!--.*?-->/gs, '');
+
+  let text = current.textRaw;
+
+  // Determine if the current item is a return statement
+  if (RETURN_EXPRESSION.test(text)) {
+    current.name = 'return';
+    text = text.replace(RETURN_EXPRESSION, '');
+  } else {
+    text = extractPattern(text, NAME_EXPRESSION, 'name');
+  }
+
+  // Extract type and default values if present
+  text = extractPattern(text, TYPE_EXPRESSION, 'type');
+  text = extractPattern(text, DEFAULT_EXPRESSION, 'default');
+
+  // Assign the remaining text as the description after removing leading hyphens
+  current.desc = text.replace(LEADING_HYPHEN, '').trim() || undefined;
+
+  // Recursively parse nested options if a list is found within the list item
+  const optionsNode = child.children.find(child => child.type === 'list');
+  if (optionsNode) {
+    current.options = optionsNode.children.map(child =>
+      parseListItem(child, entry)
+    );
+  }
+
+  return current;
+}
+
+/**
+ * Parses stability metadata and adds it to the section.
+ * @param {import('../types.d.ts').Section} section - The section to add stability to.
+ * @param {Array} nodes - The AST nodes.
+ */
+function parseStability(section, nodes) {
+  nodes.forEach((node, i) => {
```

Comment:
As I mentioned, we should use `unified` to find the stability node. However, a section (metadata) already has a tree called stability, which includes the metadata and contents of stability. We don't need to iterate this again (adding extra work)